homeassistant/components/unifi/__init__.py

```diff
@@ -1,13 +1,32 @@
 """Support for devices connected to UniFi POE."""
+from homeassistant import config_entries
 from homeassistant.const import CONF_HOST
 from homeassistant.helpers.device_registry import CONNECTION_NETWORK_MAC
 
-from .const import CONF_CONTROLLER, CONF_SITE_ID, CONTROLLER_ID, DOMAIN
+from .const import (
+    CONF_CONTROLLER, CONF_SITE_ID, CONTROLLER_ID, DOMAIN, UNIFI_CONFIG)
 from .controller import UniFiController
+from .device_tracker import PLATFORM_SCHEMA as DEVICE_TRACKER_SCHEMA
 
 
 async def async_setup(hass, config):
     """Component doesn't support configuration through configuration.yaml."""
+    unifi_config = []
+
+    if 'device_tracker' in config:
+        for dt_conf in config['device_tracker']:
+            if dt_conf['platform'] == DOMAIN:
+                unifi_config.append(DEVICE_TRACKER_SCHEMA(dt_conf))
```

Comment:
It's a bit weird that we parse the device tracker unifi platform config here in the unifi component. Have you considered making a breaking change and moving any config under the unifi integration section?

Reply:
There will be a breaking change down the line when config entry options can replace the rest of the config. No need to bother users twice

Reply:
Ok to leave as is for the moment?

Reply:
I'm not sure. I would say that using a platform config schema in a non base component is not in line with our architecture.

Reply:
It is just a Solution allowing people to migrate without effort. Avoiding unnecessary breaking changes should be an acceptable reason

Reply:
I'd check with Paulus.

Reply:
@balloob what do you say?

Reply:
Yeah we should not do this because you're going to jump into a rabbit hole. Things like `device_tracker 2:` are valid keys in a config, but are now ignored. I prefer we make it a breaking change. It wouldn't be a breaking change twice if we can get most things configured on the root level now ?

Reply:
As soon as config entry options I want to remove this old way. How would the device tracker 2 path work? That would be a custom component config? Don't think that is a valid reason to do a breaking change. This current implementation will not be long lived.

Reply:
So we have a few paths here:

1. Import device tracker and allow configuration from device tracker platform (current implementation) (1 breaking change when config will no longer be usable)

2. Import device tracker and extra options and don't care for configuration until config entry options (1 breaking change when old config is no longer usable)

3. Move configuration to unify integration configuration replacing device_tracker configuration and allow imports and extra options (2 breaking changes, one with this PR requiring user to change configuration and one later when config entry options frontend is supported)

4. No import option, all configuration is done from config entry, no configuration from (1 breaking change nor with this PR requiring user to set config entry up without import)

Personally I'd go with 1 or 4

Reply:
Why would there be two breaking changes with 3? Why another breaking change with config entry options?

Reply:
Because I will remove support for the configuration.yaml path to give extra options

Reply:
Is that necessary?

Reply:
Of course. The configuration.yaml approach for configuring config entries will not be as flexible as adding new stuff through the options path so I see it only as extra luggage

Reply:
Wouldn't each config yaml option just be mirrored in config entry options, and then you can choose to not extend config yaml but extend config entry options?

Reply:
Sure it could. But it has the drawback that if you remove the config entry and restart it will be imported again. And it will definitely be a diff in what supported options you could set since I don't really want to support the configuration.yaml way, so better to just have it removed when config entry options are available

Reply:
I don't understand what the drawback is. That sounds like how it should work if using config yaml for configuration.

The tech debt is always a valid concern. 🤷‍♂️

I think you should pick the option that fits your intentions best but is still in line with our architecture.

Reply:
@MartinHjelmare I've rewritten it now into something more in line with what you've requested